assets/mechanics.js

```diff
@@ -105,18 +108,23 @@ function update() {
     preventGoingThroughBottom(); // conditions wen ball hit bottem of canvis
     /// only needed for ground
     floors.forEach((element) => {
-      element.floorcollision(); // these condition make sure thea are repeated in this function
-      //   console.log(obstacle);
-      element.movefloorUp();
+
       
-      if (element.floorY < 80) {
-        floors.shift()
+      if (element.floorY < 80 && element.deleted == false) {
```

Comment:
If an element is lower than the threshold and has not been deleted yet.

Then we delete it (mark it for removal... but we don't actually remove it... This is a problem!)